homeassistant/components/onkyo/media_player.py

```diff
@@ -115,395 +196,350 @@
 SERVICE_SELECT_HDMI_OUTPUT = "onkyo_select_hdmi_output"
 
 
-def determine_zones(receiver):
-    """Determine what zones are available for the receiver."""
-    out = {"zone2": False, "zone3": False}
-    try:
-        _LOGGER.debug("Checking for zone 2 capability")
-        receiver.raw("ZPW")
-        out["zone2"] = True
-    except ValueError as error:
-        if str(error) != TIMEOUT_MESSAGE:
-            raise error
-        _LOGGER.debug("Zone 2 timed out, assuming no functionality")
-    try:
-        _LOGGER.debug("Checking for zone 3 capability")
-        receiver.raw("PW3")
-        out["zone3"] = True
-    except ValueError as error:
-        if str(error) != TIMEOUT_MESSAGE:
-            raise error
-        _LOGGER.debug("Zone 3 timed out, assuming no functionality")
+@callback
+def _parse_onkyo_tuple(value):
+    """Parse a value returned from the eiscp library into a tuple."""
+    if isinstance(value, str):
+        return value.split(",")
+
+    return value
+
+
+@callback
+def _tuple_get(tup, index, default=None):
+    """Return a tuple item at index or a default value if it doesn't exist."""
+    return (tup[index : index + 1] or [default])[0]
```

Comment:
Please see the comment about handling these details in the pypi package.

Reply:
The package now returns a tuple instead of a string of comma separated values if there are multiple values, so we can get rid of the _parse_onkyo_tuple function.